codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -260,6 +272,22 @@ public final DefaultStream parent() {
         }
 
         @Override
+        public final boolean isViable() {
+            return state != CLOSED;
+        }
+
+        @Override
+        public final boolean hasViableDescendant() {
```

Comment:
Lets handle the `Do we need 2 methods? Could isViable be recursive?` comment here too.

We may be able to do this but I'm of the mindset that it is makes sense to have them separated.  For example there are places (takeChild, removeChild) where we don't want to recursively check the subtree and only want to check the immediate node. WDYT?